README.md

```diff
@@ -5,7 +5,17 @@ For the moment, this repository only contains boilerplate from Spring Boot, as w
 
 `checkstyle` is used for linting. We adhere to the rules defined in the [Google style guide](https://google.github.io/styleguide/javaguide.html).
 
-For formatting java files, it is possible to use [google-java-format](https://github.com/google/google-java-format). To format all java files, run `java -jar google-java-format-1.7-all-deps.jar -i $(git ls-files|grep \.java$)`.
+## Local setup
+### Automatically format java files before commit.
+To format java files, we use [google-java-format](https://github.com/google/google-java-format).
+- Download google-java-format-1.7-all-deps.jar from https://github.com/google/google-java-format/releases
+- Move google-java-format-1.7-all-deps.jar at the root of the repository
+- Run `cp ./.git/hooks/pre-commit.sample ./.git/hooks/pre-commit`
+- Add the following lines to the top of the pre-commit file:
+
+`java -jar google-java-format-1.7-all-deps.jar -i $(git ls-files|grep \.java$)`
+
+`git add $(git ls-files|grep \.java$)`
```

Comment:
Does this mean all `.java` files will be added automatically right before committing? Are we sure we want to recommend this practice?
For instance, say I'm working on two very different features in `FeatureOne.java` and `FeatureTwo.java`. Wouldn't I want to have two distinct commits for them?

Reply:
This could also cause issues if we have some local file that we don't want to commit, `LocalMainJustForTests.java`.